test/langtools/jdk/internal/shellsupport/doc/SinceCheckerHelper.java

```diff
@@ -0,0 +1,317 @@
+import com.sun.source.util.JavacTask;
+import com.sun.tools.javac.code.Flags;
+import com.sun.tools.javac.code.Symbol;
+import jdk.internal.shellsupport.doc.JavadocHelper;
+
+import javax.lang.model.element.*;
+import javax.lang.model.util.ElementFilter;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+import javax.tools.*;
+import java.io.IOException;
+import java.lang.Runtime.Version;
+import java.net.URI;
+import java.nio.file.*;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class SinceCheckerHelper {
+
+    //these are methods that were preview in JDK 13 and JDK 14, before the introduction
+    //of the @PreviewFeature
+    static final Set<String> LEGACY_PREVIEW_METHODS = Set.of(
+            "method:java.lang.String:stripIndent:()",
+            "method:java.lang.String:translateEscapes:()",
+            "method:java.lang.String:formatted:(java.lang.Object[])");
+
+    static final int JDK9 = 9, JDK23 = 23;
+    static final String JDK13 = "13",JDK14="14";
+
+    // only one hashmap is enough for now
+    public static Map<String, IntroducedIn> classDictionary = new HashMap<>();
+    public JavaCompiler tool;
+    StringBuilder sb = new StringBuilder("");
+
+    public SinceCheckerHelper() {
+        tool = ToolProvider.getSystemJavaCompiler();
+        for (int i = JDK9; i <= JDK23; i++) {
```

Comment:
We should not go to JDK 23, we should go to `Runtime.version().feature()` or something similar. I.e. when a new JDK is started, this test should automatically start to use it.

As for the start, we can keep 9 for now, but we should be prepared to go to `Runtime.version().feature() - 1` or something similar. As we mostly care about mistakes in the current version.